torch/_functorch/deprecated.py

```diff
@@ -0,0 +1,107 @@
+import torch._functorch.vmap as _vmap_impl
+import torch._functorch.eager_transforms as _impl
+import torch._functorch.make_functional as _nn_impl
+import textwrap
+import warnings
+
+"""
+The APIs in this file are exposed as `functorch.*`. They are thin wrappers
+around the torch.func.* APIs that have deprecation warnings -- we're trying
+to move people to the torch.func.* equivalents.
+"""
+
+def get_warning(api, new_api=None, replace_newlines=False):
+    if new_api is None:
+        new_api = f'torch.func.{api}'
+    warning = (
+        f"We've integrated functorch into PyTorch. As the final step of the \n"
+        f"integration, functorch.{api} is deprecated as of PyTorch \n"
+        f"2.0 and will be deleted in a future version of PyTorch >= 2.3. \n"
+        f"Please use {new_api} instead; see the PyTorch 2.0 release notes \n"
+        f"and/or the torch.func migration guide for more details."
+    )
+    if replace_newlines:
+        warning = warning.replace("\n", "")
+    return warning
+
+
+def warn_deprecated(api, new_api=None):
+    warning = get_warning(api, new_api, replace_newlines=True)
+    warnings.warn(warning)
```

Comment:
You might want to pass in stacklevel=2 so that the line number in the warning points to `functorch.vmap()` call and not `warn_deprecated('vmap', 'torch.vmap')`

Reply:
Good point